persistence/rest/src/main/java/org/infinispan/persistence/rest/configuration/RestStoreConfigurationSerializer.java

```diff
@@ -0,0 +1,52 @@
+package org.infinispan.persistence.rest.configuration;
+
+import javax.xml.stream.XMLStreamException;
+
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.configuration.serializing.AbstractStoreSerializer;
+import org.infinispan.configuration.serializing.ConfigurationSerializer;
+import org.infinispan.configuration.serializing.XMLExtendedStreamWriter;
+
+/**
+ * RestStoreConfigurationSerializer.
+ *
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+public class RestStoreConfigurationSerializer extends AbstractStoreSerializer implements ConfigurationSerializer<RestStoreConfiguration> {
+
+   @Override
+   public void serialize(XMLExtendedStreamWriter writer, RestStoreConfiguration configuration) throws XMLStreamException {
+      AttributeSet attributes = configuration.attributes();
+      writer.writeStartElement(Element.REST_STORE);
+      attributes.write(writer, RestStoreConfiguration.KEY2STRING_MAPPER, Attribute.KEY_TO_STRING_MAPPER);
```

Comment:
you have to write all the attributes otherwise the common store attributes aren't written. If you do ```attributes.write(writer);```, you have to configure the ```HOST```and ```PORT``` auto-persist to false.